mcstatus/server.py

```diff
@@ -48,22 +52,48 @@ def port(self) -> int:
 
     @classmethod
     def lookup(cls, address: str, timeout: float = 3) -> Self:
-        """Parses the given address and checks DNS records for an SRV record that points to the Minecraft server.
+        """Mimics minecraft's server address field.
+
+        :param str address: The address of the Minecraft server, like `example.com:19132`
+        :param float timeout: The timeout in seconds before failing to connect.
+        """
+        addr = Address.parse_address(address, default_port=None)
+        return cls(addr.host, addr.port, timeout=timeout)
```

Comment:
Is it possible to reword this here and inherit the docstring in child classes? There's a typo in mimics (mimcs) down below.
```suggestion
        """Mimics behavior of Minecraft's server address field when adding a multiplayer server.

        :param str address: The address of the Minecraft server, like `example.com:19132`
        :param float timeout: The timeout in seconds before failing to connect.
        """
        raise NotImplementedError
```